Feature/sc 23 homepage

tsisko20 opened the pull request:
Added all design components relevant to the home page to the project, and implemented design for home page. Home page consists of one button "Connect" that when pressed it changes its design to yellow color and text to "Cancel". Idea of button design change when clicked is to tell user that he is in RFID scanning state and he can cancel it. Hamburger menu is still not implemented, but icon for its button is located on the page.

tsisko20 committed:
Added fonts, images and custom design element for button

tsisko20 committed:
Implemented design of home page.

tsisko20 committed:
Implemented simple logic of button Connect

mkajic20 commented on Software/app/src/main/java/org/foi/air/smartcharger/MainActivity.kt:
```diff
@@ -2,10 +2,37 @@ package org.foi.air.smartcharger
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        val button = findViewById<Button>(R.id.button1)
```
I don't think button1 is a good name for a button, please change the name so it's use is described.

mkajic20 commented on Software/app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,121 @@
+            app:srcCompat="@drawable/bg_waves" />
+
+        <TextView
+            android:id="@+id/textView4"
```
Please change textView4 to a different id that describes it's purpose

mkajic20 commented on Software/app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,121 @@
+            app:layout_constraintTop_toBottomOf="@+id/cardIcon" />
+
+        <Button
+            android:id="@+id/button1"
```
Same comment as the earlier one about the button

mkajic20 reviewed (changes requested):
Everything seems good, except for some names, I left comments on what I think you should change.

irosic20 commented on Software/app/src/main/java/org/foi/air/smartcharger/MainActivity.kt:
It would be good to write the types that these variables hold, like "txtInstruction" and i would use a better name for button variable that would correctly send message about its functionality, like "btnScanningState". I would also advise using view binding for finding your xml components inside the code.

irosic20 commented on Software/app/src/main/res/layout/activity_main.xml:
I recommend using better names with prefix of which component they are representing, for Button it would be "bt", TextView would be "tv"... Also you should extract strings from xml code for better maintainability for multiple languages.

irosic20 reviewed (changes requested):
Overall good code, I've left few comments about changes you should implement.

tsisko20 committed:
Added View Binding, fixed variable naming convention, and changed string literals into string resources

tsisko20 commented:
Thank you for your reviews! I changed the naming convention, replaced string literals with string resources and added View Binding. Please check if it is good now.

mkajic20 reviewed (approved):
Everything is good now.

irosic20 reviewed (approved):
Great job, code looks much better now. Keep up the work!
